examples/collector/README.md

```diff
@@ -0,0 +1,25 @@
+# Unified Runtime XPTI collector example
+
+This example demonstrates how to use the tracing layer integrated into
+Unified Runtime. It leverages XPTI to intercept all UR calls in the traced
+application and prints information about each function to standard output.
+
+## Running the example
+
+To run the example, you will need to build Unified Runtime with
+tracing enabled (`-DUR_ENABLE_TRACING=ON`), link it with your application,
+and then set appropriate environment variables for the collector (`XPTI_SUBSCRIBERS`) and the
+XPTI dispatcher (`XPTI_FRAMEWORK_DISPATCHER`) to be loaded and enabled (`XPTI_TRACE_ENABLE`).
+You will also need to setup your chosen adapter (`UR_ADAPTERS_FORCE_LOAD`).
+
+For example, to run the `hello_world` example with the collector, use:
+
+```
+$ mkdir build
+$ cd build
+$ cmake .. -DUR_ENABLE_TRACING=ON
+$ make
+$ UR_ADAPTERS_FORCE_LOAD=./lib/libur_null.so XPTI_TRACE_ENABLE=1 XPTI_FRAMEWORK_DISPATCHER=./lib/libxptifw.so XPTI_SUBSCRIBERS=./lib/libcollector.so ./bin/hello_world
```

Comment:
Can you set the XPTI-related env vars in CMakeLists during configuration or building? So the user won't have to type it all himself? After all, if I configure UR with UR_ENABLE_TRACING=ON then I most probably want to use this feature.

Reply:
I'm not sure what you are asking. These are runtime variables that the xpti proxy library and the dispatcher look for during initialization. I cannot bake them into a binary, nor should they be.
Having all the environment variables spelled out explicitly helps the user see how to use the tracing functionality without anything being hidden by scripts.